app/components/uploaderbutton/uploaderbutton.js

```diff
@@ -0,0 +1,48 @@
+
+/**
+ * Copyright (c) 2016, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ */
+
+import React, { Component } from 'react'
+
+import logoSrc from './images/T-logo-dark-512x512.png';
+
+export default class UploaderButton extends Component {
+  static propTypes = {
+    buttonUrl: React.PropTypes.string.isRequired,
+    onClick: React.PropTypes.func,
```

Comment:
My thinking was that maybe you'd want a button to link somewhere, without having an `onClick` handler (for tracking or a callback or whatever). Just an old school link. :) But I'm not opposed to requiring it, especially if that's our convention!